### ISSUE TITLE
Bug: no-magic-numbers error message for invalid BigInt in configuration

### ISSUE THREAD

**Issue by icecream17** [eslint/eslint#15579]
### Environment

Node version: v17.4.0
npm version: v8.4.1
Local ESLint version: v8.8.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19042

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```json
{
   "extends": [
      "hardcore/ts"
   ],
   "parserOptions": {
      "project": "./tsconfig.json",
      "sourceType": "module"
   },
   "rules": {
      "@typescript-eslint/no-magic-numbers": [
         "error",
         { "ignore": [ "0.5" ] }
      ]
   }
}
```
</details>

```js
// It fails when parsing the eslint rule
```

### What did you expect to happen?

Eslint ignores the magic number 0.5

(Note: The suggested change is to change the error message)

### What actually happened?

```shell
PS C:\Users\Steven\Documents\code\js\test\ts-hardcore> pnpm test

> ts-hardcore@0.1.0 test C:\Users\Steven\Documents\code\js\test\ts-hardcore
> eslint index.ts



ESLint: 8.8.0

SyntaxError: Error while loading rule '@typescript-eslint/no-magic-numbers': Cannot convert 0. to a BigInt
Occurred while linting C:\Users\Steven\Documents\code\js\test\ts-hardcore\index.ts
    at BigInt (<anonymous>)
    at normalizeIgnoreValue (C:\Users\Steven\Documents\code\js\test\ts-hardcore\node_modules\.pnpm\eslint@8.8.0\node_modules\eslint\lib\rules\no-magic-numbers.js:24:16)
    at Array.map (<anonymous>)
    at Object.create (C:\Users\Steven\Documents\code\js\test\ts-hardcore\node_modules\.pnpm\eslint@8.8.0\node_modules\eslint\lib\rules\no-magic-numbers.js:83:44)
    at create (C:\Users\Steven\Documents\code\js\test\ts-hardcore\node_modules\.pnpm\@typescript-eslint+eslint-plugin@5.10.2_2595c2126aec4d4b6e944b931dabb4c2\node_modules\@typescript-eslint\eslint-plugin\dist\rules\no-magic-numbers.js:64:32)
    at Object.create (C:\Users\Steven\Documents\code\js\test\ts-hardcore\node_modules\.pnpm\@typescript-eslint+utils@5.10.2_eslint@8.8.0+typescript@4.5.5\node_modules\@typescript-eslint\utils\dist\eslint-utils\RuleCreator.js:42:20)
    at createRuleListeners (C:\Users\Steven\Documents\code\js\test\ts-hardcore\node_modules\.pnpm\eslint@8.8.0\node_modules\eslint\lib\linter\linter.js:915:21)
    at C:\Users\Steven\Documents\code\js\test\ts-hardcore\node_modules\.pnpm\eslint@8.8.0\node_modules\eslint\lib\linter\linter.js:1097:31
    at Array.forEach (<anonymous>)
    at runRules (C:\Users\Steven\Documents\code\js\test\ts-hardcore\node_modules\.pnpm\eslint@8.8.0\node_modules\eslint\lib\linter\linter.js:1034:34)
 ELIFECYCLE  Test failed. See above for more details.
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

https://github.com/eslint/eslint/blob/355b23d0c4e050be4e53292f552a47c10ec6e00e/lib/rules/no-magic-numbers.js#L24

It seems like this rule expects a bigint if the argument is a string. So that fixes the problem for me.

I think a clearer error message might be `Expected string to be bigint, got "0.5"`

**Comment by mdjermanovic** [comment]
Hi @icecream17, thanks for the issue!

I'm getting a different error with the core `no-magic-numbers` rule.

config:

```json
{
    "rules": {
       "no-magic-numbers": [
          "error",
          { "ignore": [ "0.5" ] }
       ]
    }
 }
```

error:

```
Oops! Something went wrong! :(

ESLint: 8.8.0

Error: .eslintrc.json:
        Configuration for rule "no-magic-numbers" is invalid:
        Value "0.5" should be number.
        Value "0.5" should match pattern "^[+-]?(?:0|[1-9][0-9]*)n$".
        Value "0.5" should match some schema in anyOf.
```

This error message is coming from Ajv validations, and is based on the following schema for the `ignore` property:

https://github.com/eslint/eslint/blob/355b23d0c4e050be4e53292f552a47c10ec6e00e/lib/rules/no-magic-numbers.js#L51-L60

We can't control schema validation messages, and this one seems descriptive (`Value "0.5" should be number` looks like the correct suggestion to fix the problem), so I think this works well.

-----

> ```json
> {
>    "extends": [
>       "hardcore/ts"
>    ],
>    "parserOptions": {
>       "project": "./tsconfig.json",
>       "sourceType": "module"
>    },
>    "rules": {
>       "@typescript-eslint/no-magic-numbers": [
>          "error",
>          { "ignore": [ "0.5" ] }
>       ]
>    }
> }
> ```

You are actually using [`@typescript-eslint/no-magic-numbers`](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-magic-numbers.md) rule, which loads the core `no-magic-numbers` rule and extends it with several typescript-specific options.

What seems wrong is that your configuration successfully passes schema validations and then fails inside the rule which doesn't expect invalid values because they shouldn't pass schema validations. I believe that `@typescript-eslint/no-magic-numbers` doesn't correctly merge the original schema with the added options, and that therefore it exports a wrong schema. In particular, it looks like this [`{ ...baseRule.meta.schema }`](https://github.com/typescript-eslint/typescript-eslint/blob/420936274603d8fee0631cdb3b58874cb3cc59d7/packages/eslint-plugin/src/rules/no-magic-numbers.ts#L12) should be `{ ...baseRule.meta.schema[0] }`. Can you please report this issue to typescript-eslint?

https://github.com/typescript-eslint/typescript-eslint

I'm closing this issue since it isn't related to core ESLint rules.
